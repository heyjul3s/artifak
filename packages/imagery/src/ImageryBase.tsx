import { StyledComponent } from 'styled-components';
import { system } from 'styled-system';
import { createStyledComponent } from '@artifak/component-generator';
import { ImgComponent } from './typings';

const imageStylesConfig = system({
  objectFit: {
    property: 'objectFit'
  },
  objectPosition: {
    property: 'objectPosition'
  }
});

export const ImageryBase: StyledComponent<
  'img',
  any,
  ImgComponent.Props,
  any
> = createStyledComponent<Partial<ImgComponent.Props>>(
  [imageStylesConfig],
  { display: 'block', width: '100%', height: 'auto' },
  'img'
);
