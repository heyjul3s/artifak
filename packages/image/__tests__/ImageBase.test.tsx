import React from 'react';
import { render } from '@testing-library/react';
import { ImageryBase } from '../ImageryBase';

describe('ImageryBase', () => {
  it('should render', () => {
    const { asFragment } = render(<ImageryBase />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
