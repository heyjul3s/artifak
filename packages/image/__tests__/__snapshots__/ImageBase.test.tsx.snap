// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageryBase should render 1`] = `
<DocumentFragment>
  <img
    class="artifak__sc-6om8fa-0 brhrtq"
  />
</DocumentFragment>
`;
